File: CardsCreator.DomainModel/Card.cs
﻿using System;

namespace CardsCreator.DomainModel
{
    public class Card
    {
        public Card(LanguageType sideOneLang, LanguageType sideTwoLang, string wordOne, string wordTwo)
        {
            SideOne = new Side { LanguageType = sideOneLang, Text = wordOne };
            SideTwo = new Side { LanguageType = sideTwoLang, Text = wordTwo };
        }

        public Side SideOne { get; set; }

        public Side SideTwo { get; set; }

        public bool IsCompleted => !string.IsNullOrWhiteSpace(SideTwo.Text) && !string.IsNullOrWhiteSpace(SideOne.Text) && 
                                    SideOne.LanguageType != LanguageType.Undefined && SideTwo.LanguageType != LanguageType.Undefined;
    }

    public class Side
    {
        public LanguageType LanguageType { get; set; }
        public string Text { get; set; }
    }
}

File: CardsCreator/ClientApp/vue-clientapp/src/models/card.js
import LanguageType from './languageType.js'

export default class Card {
   constructor(sideOneText = '', sideTwoText = '', 
      langTypeOne = LanguageType.Undefined, 
      langTypeTwo = LanguageType.Undefined) {


      this.SideOne = new Side(langTypeOne, sideOneText)
      this.SideTwo = new Side(langTypeTwo, sideTwoText)
      this.GoToPrint = true; 
   }
   get IsCompleted() {
      return this.SideOne.Text.trim().length > 0 && this.SideTwo.Text.trim().length > 0;
   }
}

class Side {
   constructor(languageType, text) {
      this.LanguageType = languageType;
      this.Text = text
   }
}